pkg/scannerv4/mappers/mappers.go

```diff
@@ -633,22 +674,40 @@ type severityValues struct {
 	v2Score  float32
 	v3Vector string
 	v3Score  float32
+	source   v4.VulnerabilityReport_Vulnerability_CVSS_Source
+	url      string
 }
 
-// severityAndScores returns the severity and scores information out of a
-// ClairCore vulnerability. The returned information is dependent on the
-// underlying updater.
-func severityAndScores(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) (severityValues, error) {
+// severityAndCvssMetrics processes the severity and CVSS scores
+// for a given vulnerability based on the updater type (RHEL, NVD, OSV, etc.).
+// The returned data includes CVSS metrics, ACS preferred severity values, and potential errors
+// encountered during the calculation process for each source.
+func severityAndCvssMetrics(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) ([]*v4.VulnerabilityReport_Vulnerability_CVSS, severityValues, error) {
 	ctx = zlog.ContextWithValues(ctx,
-		"component", "mappers/severityAndScores",
+		"component", "mappers/severityAndCvssMetrics",
 		"updater", vuln.Updater,
 		"vuln_id", vuln.ID,
 		"vuln_name", vuln.Name,
 	)
-
+	var cvssMetrics []*v4.VulnerabilityReport_Vulnerability_CVSS
+	rhelSev, rhelErr := rhelSeverityAndScores(vuln)
+	if rhelErr == nil {
+		cvssMetrics = toCVSSMetrics(ctx, rhelSev, cvssMetrics)
+	}
+	nvdSev, nvdErr := nvdSeverityAndScores(vuln, nvdVuln)
+	if nvdErr == nil {
+		cvssMetrics = toCVSSMetrics(ctx, nvdSev, cvssMetrics)
+	}
+	unknwonSev, unknownErr := cvssVector(vuln.Severity)
+	if unknownErr == nil {
+		// source needs to be specified for OSV as cvssVector() method only provides severity vector
+		unknwonSev.source = v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_UNKNOWN
```

Comment:
hmm I wonder if we should make an OSV-specific function, so it's not unknown. Also, when would it be unknown? If it's not Red Hat or OSV, then it's NVD. Though, our manual vulns may be considered unknown, but I think they currently match NVD, and I'd be tempted to claim they're from NVD. Then again, that may also be false, and we should create a separate StackRox  source or something

Reply:
what do you think of this?
https://github.com/stackrox/stackrox/pull/12612/commits/f3f33e63130ade373113d55ff768d60ede980b67
Also probably we can convert vector to score in `cvssVector()` (not included in the commit)
And I am ok with add a new source for manual updater

Reply:
I have changed unknwonSev to `undecidedSev` and then update the sev source based on updater name.